src/thing_list.c

```diff
@@ -690,7 +712,8 @@ long anywhere_thing_filter_call_bool_filter(const struct Thing *thing, MaxTngFil
 {
     if ((param->class_id == -1) || (thing->class_id == param->class_id))
     {
-        if ((param->model_id == -1) || (thing->model == param->model_id))
+        if((is_creature_match(thing, param->model_id) && param->class_id == TCls_Creature) 
```

Comment:
maybe a bit cleaner to first do the check on creature class_id and then is_creature_match, would give same result but bit more readable